### PR TITLE
Travis

### PR DESCRIPTION
**Pull request by m-zacharias**
This PR adds .travis.yml for CI with travis-ci.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - sudo apt-get install -qq build-essential
+  - gcc --version && g++ --version # 4.8
+  - apt-cache search nvidia-*
+  - sudo apt-get install -qq nvidia-common
+  - sudo apt-get install -qq nvidia-cuda-dev nvidia-cuda-toolkit # 5.5
+  - sudo apt-get install -qq libboost-dev # 1.54.0
+#  - sudo find /usr/ -name libcuda*.so
+
+after_script:
+#  - ls -halR $INSTALL_DIR
```
Comment (m-zacharias) [inline .travis.yml]: Newline missing at EOF.

**Comment by m-zacharias** [comment]
In .travis.yml:
- slim out boost installations
- clean up
- look for other unnecessary stuff
- turn on -Wextra
- maybe turn on -Werror
- squash

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#  - sudo dpkg --get-selections | grep hold || { echo "All packages OK."; }
+  - sudo apt-get install -q -y cmake-data cmake
+  - cmake --version
+  
+  - sudo apt-get install -qq build-essential
+  - gcc --version && g++ --version # 4.8
+#  - sudo apt-get install -qq nvidia-common
+#  - sudo apt-get install -qq nvidia-cuda-dev nvidia-cuda-toolkit # 5.5
+  
+  - travis_retry wget http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1404/x86_64/cuda-repo-ubuntu1404_${CUDA}_amd64.deb
```
Comment (ax3l) [inline .travis.yml]: please use [caching](https://docs.travis-ci.com/user/caching/), see [this example](https://github.com/ComputationalRadiationPhysics/cupla/blob/dev/.travis.yml)

Comment (m-zacharias) [inline .travis.yml]: i do understand that caching will stuff persistent between builds. but in the example everthing is built from source. is that somehow the point, do we regard it as better practice?

Comment (m-zacharias) [inline .travis.yml]: wait, i think i got it. there's still packages being installed but they're circumventing disabled "sudo apt-get ..." via "addons: apt: ...". Right?

**Comment by m-zacharias** [comment]
Got everything done and squashed stuff. Going to merge.

**Pull request merged**